tests/merchOrder.test.ts

```diff
@@ -590,6 +590,44 @@ describe('merch orders', () => {
       .rejects.toThrow(`Not allowed to order: ${[option.uuid]}`);
   });
 
+  test('order route accurately verifies monthly and lifetime limits', async () => {
+    const conn = await DatabaseConnection.get();
+    const optionMetadataType = faker.datatype.hexaDecimal(10);
+    const option1 = MerchFactory.fakeOptionWithType(optionMetadataType);
+    const option2 = MerchFactory.fakeOptionWithType(optionMetadataType);
+    const item = MerchFactory.fakeItem({
+      options: [option1, option2],
+      monthlyLimit: 1,
+      lifetimeLimit: 1,
+    });
+    const orderPickupEvent = MerchFactory.fakeFutureOrderPickupEvent();
+    const member = UserFactory.fake({
+      accessType: UserAccessType.STANDARD,
+      credits: 1 * option1.price + 1 * option2.price,
+    });
+
+    await new PortalState()
+      .createUsers(member)
+      .createOrderPickupEvents(orderPickupEvent)
+      .createMerchItem(item)
+      .write();
+
+    member.reload();
+
+    const emailService = mock(EmailService);
+    when(emailService.sendOrderConfirmation(anything(), anything(), anything()))
+      .thenResolve();
+
+    const merchController = ControllerFactory.merchStore(conn, instance(emailService));
+    const placeMerchOrderRequest = {
+      order: [{ option: option1.uuid, quantity: 1 }, { option: option2.uuid, quantity: 1 }],
+      pickupEvent: orderPickupEvent.uuid,
+    };
+    // console.log(placeMerchOrderRequest.order)
+    // await merchController.placeMerchOrder(placeMerchOrderRequest,member);
+    await expect(merchController.placeMerchOrder(placeMerchOrderRequest, member)).rejects.toThrowError(HttpError);
+  });
```

Comment:
I'm still a bit skeptical on why an `HttpError` is being thrown here instead of a `UserError`. Do we know that the error being thrown is this exact error?
```ts
throw new UserError(`This order exceeds the lifetime limit for ${itemWithQuantity.item.itemName}`);
```

Reply:
I think there is some bug in Jest since when I waited for `UserError` threw an error when it calls the super constructor (effectively making it an HttpError as far as Jest understands)

Reply:
Just made a fix that looks for the specific error message so that should work better